qemu/tests/qemu_guest_agent.py

```diff
@@ -1828,6 +1828,44 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
         else:
             test.fail("Should return error info.")
 
+    @error_context.context_aware
+    def gagent_check_log(self, test, params, env):
+        """
+        Check guest agent logs.
+        Steps:
+        1) start guest-agent to record logs
+        2) issue some guest agent commands
+        3) check agent log, if those commands are recorded
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameterspy
+        :param env: Dictionary with test environment.
+        """
+        session = self._get_session(self.params, self.vm)
+        self._open_session_list.append(session)
+        self._change_bl(session)
+
+        error_context.context("Issue some common guest agent commands.",
+                              logging.info)
+        qga_cmd_list = ["guest-get-time", "guest-file-open",
+                        "guest-file-write", "guest-file-read"]
+        self.gagent.get_time()
+        tmp_file = params["tmp_file"]
+        content = "hello world\n"
+        ret_handle = int(self.gagent.guest_file_open(tmp_file, mode="w+"))
+        self.gagent.guest_file_write(ret_handle, content)
+        self.gagent.guest_file_read(ret_handle)
+
+        error_context.context("Check guest agent debug logs.", logging.info)
+        get_log_cmd = params["get_log_cmd"]
+        agent_log = session.cmd_output(get_log_cmd)
+
+        for qga_cmd in qga_cmd_list:
```

Comment:
Do you care about the order of the log? From my perspective it would be more precise if you check the last line of the log immediately after take one specific action.

Reply:
the order is not important, the point is to check if the command is recorded. But your suggestion makes sense,will modify, thanks.